kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt

```diff
@@ -139,7 +141,6 @@ fun <T : Any> beEqualToUsingFields(other: T, vararg fields: KProperty<*>): Match
  * Note: Throws [IllegalArgumentException] in case [properties] parameter is not provided.
  */
 fun <T : Any> T.shouldBeEqualToIgnoringFields(other: T, vararg properties: KProperty<*>) {
```

Comment:
Do you think this method should allow specifying no fields? Equal to ignoring nothing? That would seem to be a strange way to use this method.

Reply:
Well, that's what has been [recommend](https://github.com/kotest/kotest/issues/1161#issuecomment-575724787) to me for me use case. From a strictly logical perspective this sounds alright, but I agree it feels a bit unnatural.

But if neither `shouldBeEqualToUsingFields` nor `shouldBeEqualToIgnoringFields` offer an easy way to compare *all* fields, and preferably also with an option to compare `private` fields, what's the alternative?